posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,800 @@
+- debuginfo: Fix bug in type name generation for dyn types with associated types but no other generic arguments. [#94810](https://github.com/rust-lang/rust/pull/94810)
+- async: Give predictable name to binding generated from .await expressions. [#95011](https://github.com/rust-lang/rust/pull/95011)
+- debuginfo: Fix debuginfo for Box&lt;T&gt; where T is unsized. [#95270](https://github.com/rust-lang/rust/pull/95270)
+- debuginfo: Emit ZST struct debuginfo for unit type when CPP-like debuginfo is enabled [#96316](https://github.com/rust-lang/rust/pull/96316)
+
+Surprises:
+Debuginfo just doesn't have enough test coverage, but that isn't particularly surprising.
+
+**Regarding prioritization and focus:** debugging in general is a top priority for mw & wesleywiser’s
+team.
+
+### supporting split debuginfo
+
+<!-- https://hackmd.io/@rust-compiler-team/ByXfjiXu5 -->
+
+**Progress:** The Rust project has not made any progress on this ambition since the start of the year (pnkfelix: but the free form answer somewhat contradicts this)
+
+**Goals:** As of today, we think this year’s planned goals for this ambition will be achieved in the next six months, and we think the most important parts of a solution will be available in the next six months.
+
+**Where we started:** At the start of 2022, we had completed most of our plan; our main focus was polish and getting the work into the hands of Rust users.
+
+**How it's going:** As of today, we think we have implemented a solution to the problem, but most of our work has not yet landed in hands of Rust users
+
+**Details:** There hasn't been a lot of progress on split debuginfo in 2022 because most of the big pending work that was in-progress landed in late 2021 (namely integration of [thorin](https://github.com/rust-lang/thorin) into rustc to support cross-crate Split DWARF). Since then, Split DWARF has been basically complete and just simmering on nightly; an FCP for stabilization on Linux has completed. Split debuginfo on non-DWARF platforms is already complete and stable.
+
+Progress this year has primarily been some benchmarking of Split DWARF and some work to make split debuginfo an option when bootstrapping rustc (i.e. in `config.toml`), but not a lot else.
+
+Future work is basically just stabilization of `-Csplit-debuginfo` on Linux (Split DWARF); and of the currently-default options for the other platforms (for example, `-Csplit-debuginfo=packed` on Windows requires `-Zunstable-options` despite being effectively the default if you don't specify any flags).
+
+The owner of this work, davidtwco, intends to stick with the theme of debugging and contribute to the wg-debugging working group, but has also shifted attention to diagnostic translation they see that as an interesting area where they can have impact (and because the remaining implementation tasks for split debuginfo were completed as noted above).
```

Comment:
Nit

```suggestion
The owner of this work, @davidtwco, intends to stick with the theme of debugging and contribute to the wg-debugging working group, but has also shifted attention to diagnostic translation they see that as an interesting area where they can have impact (and because the remaining implementation tasks for split debuginfo were completed as noted above).
```